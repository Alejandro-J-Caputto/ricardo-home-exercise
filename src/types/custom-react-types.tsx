import { Dispatch, SetStateAction } from "react";

export type StateSetter =  Dispatch<SetStateAction<any>>