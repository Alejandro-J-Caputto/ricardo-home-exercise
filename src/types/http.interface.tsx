import {
  ArticleResponse,
  GetUserResponse,
  SearchArticle,
  SearchResponse,
} from "./response.types";

export default interface HttpConfig {
  endpoint: string;
  params: string;
  method?: string;
  headers?: Headers;
  body?: any;
}

export type cbData = (data?: GetUserResponse | ArticleResponse | SearchResponse | SearchArticle | SearchArticle[], dataIds?:string[]) => void;
