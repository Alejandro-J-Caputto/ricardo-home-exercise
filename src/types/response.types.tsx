export type User = {
  id: string;
  name: string;
};

export type GetUserResponse = User;

export type ArticleDetails = {
  id: string;
  title: string;
  subtitle: string;
  price: number;
  descriptionHtml: string;
  imageUrl: string;
  sellerId: string;
};

export type ArticleResponse = ArticleDetails;

export type SearchArticle = {
  id: number;
  title: string;
  endDate: string;
  imageUrl: string;
  buyNowPrice: number;
};

export type SearchResponse = {
  articles: SearchArticle[];
  totalCount: number;
};
