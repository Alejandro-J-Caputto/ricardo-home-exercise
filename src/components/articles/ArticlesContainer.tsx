import { SearchArticle } from "../../types/response.types";
import ArticleItem from "./ArticleItem";

const ArticlesContainer: React.FC<{ items: SearchArticle[], isLoading:boolean }> = (props) => {
  const { items, isLoading } = props;
  return (
    <>
      {isLoading === false ? <p className="count-result">Total Count: {items.length + 1} </p> : null}

      <div className="articles-grid">
        {!items.length && !isLoading ? (
          <h2>NO RESULTS FOUND</h2>
        ) : (
          items.map((el) => <ArticleItem key={el.id} itemsContent={el} />)
        )}
      </div>
    </>
  );
};

export default ArticlesContainer;
