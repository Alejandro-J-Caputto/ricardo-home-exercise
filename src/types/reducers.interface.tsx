import { ArticleDetails, SearchArticle, User } from "./response.types";

export interface ArticleInitialState {
  articles: SearchArticle[];
  articlesById: ArticleDetails;
  sellerById: User;
  savedArticlesIDs: string[];
  selectedItemsLocalStorage: SearchArticle[];
}

export interface UIinitialState {
  loadingHTTP: boolean;
  loadingApp: boolean;
  uiTheme: boolean;
}
