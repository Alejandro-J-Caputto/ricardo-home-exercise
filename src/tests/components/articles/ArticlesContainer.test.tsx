import { shallow, configure } from "enzyme";

import "@testing-library/jest-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";
import ArticlesContainer from "../../../components/articles/ArticlesContainer";
import { SearchArticle } from "../../../types/response.types";

configure({ adapter: new Adapter() });
//@ts-ignore
expect.addSnapshotSerializer(createSerializer({ mode: "deep", noKey: true }));

describe("Tests on ArticlesContainer", () => {
  const items: SearchArticle[] = [];
  let isLoading = true;
  const wrapper = shallow(
    <ArticlesContainer items={items} isLoading={isLoading} />
  );

  test("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should not display the <p>Total Count<p/>", () => {
    //@ts-ignore

    isLoading = true;
    const totalCount = wrapper.find("p");
    expect(totalCount.exists()).toBe(false);
  });
});
