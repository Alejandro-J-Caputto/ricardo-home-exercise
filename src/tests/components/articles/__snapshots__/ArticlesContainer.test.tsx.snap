// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on ArticlesContainer should match the snapshot 1`] = `
<Fragment>
  <div
    className="articles-grid"
  />
</Fragment>
`;
